### PR TITLE
cubelib, cubew: Update to 4.8.2 and enable Clang based builds

### PR DESCRIPTION
**Pull request by hmartinez82**

### CODE REVIEW

File: mingw-w64-cubelib/PKGBUILD [hunk 1]
```diff
@@ -4,27 +4,34 @@
...
-mingw_arch=('mingw32' 'mingw64' 'ucrt64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
 url="http://www.scalasca.org/software/cube-4.x/download.html"
 license=('BSD')
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          "${MINGW_PACKAGE_PREFIX}-qt5-base")
-makedepends=("${MINGW_PACKAGE_PREFIX}-cc") 
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             $([[ ${MINGW_PACKAGE_PREFIX} != *-clang-* ]] || echo "${MINGW_PACKAGE_PREFIX}-flang"))
```
Comment (MehdiChinoune) [inline mingw-w64-cubelib/PKGBUILD:18]:
```suggestion
             $([[ ${MSYSTEM} == CLANG32 ]] || echo "${MINGW_PACKAGE_PREFIX}-fc"))
```
Is there a reason why not using gfortran on GCC environments?

File: mingw-w64-cubew/PKGBUILD [hunk 2]
```diff
@@ -4,27 +4,34 @@
...
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
 url="http://www.scalasca.org/software/cube-4.x/download.html"
 license=('BSD')
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-zlib")
-makedepends=("${MINGW_PACKAGE_PREFIX}-cc")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             $([[ ${MINGW_PACKAGE_PREFIX} != *-clang-* ]] || echo "${MINGW_PACKAGE_PREFIX}-flang"))
```
Comment (MehdiChinoune) [inline mingw-w64-cubew/PKGBUILD:17]:
```suggestion
             "${MINGW_PACKAGE_PREFIX}-fc")
```

File: mingw-w64-cubelib/PKGBUILD (hunk 1 above)
Comment (hmartinez82) [reply mingw-w64-cubelib/PKGBUILD:18]: Hum.. I didn't want to touch the existing behavior for the MINGW* envs. I was just trying to add CLANG64 and CLANGARM64. Should I add gfortran as a dep for the MINGW* envs?

File: mingw-w64-cubew/PKGBUILD (hunk 2 above)
Comment (hmartinez82) [reply mingw-w64-cubew/PKGBUILD:17]: Ok, I see, fc is a metapackage. I did not think of that. I'll make this change

File: mingw-w64-cubelib/PKGBUILD (hunk 1 above)
Comment (MehdiChinoune) [reply mingw-w64-cubelib/PKGBUILD:18]: just `${MINGW_PACKAGE_PREFIX}-fc` which is gfortran on GCC envs.

**Comment by hmartinez82** [comment]
Can someone kickoff the MINGW32 build again?

**Comment by MehdiChinoune** [comment]
> Can someone kickoff the MINGW32 build again?

Until all jobs finish.

**Pull request merged**